x-pack/platform/plugins/shared/cases/public/components/markdown_editor/eui_form.tsx

```diff
@@ -90,15 +116,21 @@ export const MarkdownEditorForm = React.memo(
             label={field.label}
             labelAppend={field.labelAppend}
           >
-            <MarkdownEditor
-              ref={ref}
-              ariaLabel={idAria}
-              editorId={id}
-              onChange={field.setValue}
-              value={field.value}
-              disabledUiPlugins={disabledUiPlugins}
-              data-test-subj={`${dataTestSubj}-markdown-editor`}
-            />
+            {hasFilesContext ? (
+              <PastableMarkdownEditor
+                {...editorBaseProps}
+                ref={ref}
+                field={field}
+                caseId={caseId}
+              />
+            ) : (
+              <MarkdownEditor
+                {...editorBaseProps}
+                ref={ref}
+                onChange={field.setValue}
+                value={field.value}
+              />
+            )}
```

Comment:
It would be preferably to have this logic be encapsulated inside the markdown editor. The form should not care about the internals of the editor.

Reply:
I'll see about pushing the complexity further down 👍